modules/n1ql/pages/n1ql-language-reference/recursive.adoc

```diff
@@ -0,0 +1,315 @@
+
+* ORDER BY, LIMIT, and OFFSET clauses are not allowed in the SELECT statement in the subquery used to define the anchor and the recursive clause.
+
+* The DISTINCT quantifier is not allowed in anchor and recursive clauses.
+
+* GROUP BY, WINDOW, and AGGREGATE functions are not allowed in recursive clauses.
+
+* OUTER JOINS are not allowed in recursive clauses because they can lead to potential infinite recursion.
+
+* Recursive clauses do not support NEST and UNNEST clauses.
+
+* If there is no UNION/UNION ALL separation for the recursive CTE, the query defaults to a normal CTE.
+
+* A syntax error is returned when optional subclauses are used without the RECURSIVE keyword.
+
+* In general, recursion is also limited by:
+** The logic in the recursive statement.
+** The stop in the options argument.
+** Breaching the request timeout, if configured.
+** Breaching the request memory quota, if configured.
+** Exceeding the implicit document limit (10000) when a memory quota is not in use and when no explicit document limit is set in the options.
+** Exceeding the implicit level limit (1000) when the level option is not in use.
+
+[#examples_section]
+== Examples
+
+The following examples follow linear recursion.
+Only one recursive reference is allowed in the FROM clause.
+Self-joins or set-ops are not allowed in the recursive reference.
+
+.Example of simple recursive referencing
+====
+[source,sqlpp]
+----
+WITH RECURSIVE cte AS (
+    SELECT 1 AS r 
+        UNION 
+    SELECT cte.r+1 AS r 
+    FROM cte 
+    WHERE cte.r<4 
+) 
+SELECT cte.r FROM cte;
+----
+
+.Results
+[source,json]
+----
+[
+  {
+    "r": 1
+  },
+  {
+    "r": 2
+  },
+  {
+    "r": 3
+  },
+  {
+    "r": 4
+  }
+]
+----
+====
+
+.Example combining recursive and non-recursive CTEs with the WITH clause
+====
+[source,sqlpp]
+----
+WITH RECURSIVE cte AS (SELECT 1 r) ,
+    rcte AS ( 
+        SELECT cte.r FROM cte 
+            UNION 
+        SELECT rcte.r+2 r FROM rcte WHERE rcte.r<7
+    )
+SELECT * FROM rcte;
+----
+
+.Results
+[source,json]
+----
+[
+  {
+    "rcte": {
+      "r": 1
+    }
+  },
+  {
+    "rcte": {
+      "r": 3
+    }
+  },
+  {
+    "rcte": {
+      "r": 5
+    }
+  },
+  {
+    "rcte": {
+      "r": 7
+    }
+  }
+]
+----
+====
+
+.Example finding all possible routes leaving from the TLV airport and going to 2 other airports
+====
+[source,sqlpp]
+----
+WITH RECURSIVE recroute AS (
+    SELECT sourceairport, destinationairport, 0 as depth,
+    [ sourceairport, destinationairport ] as route
+    FROM `travel-sample`.inventory.route 
+    WHERE sourceairport="TLV"
+        UNION
+    SELECT r.sourceairport, r.destinationairport, 
+    ARRAY_APPEND ( recroute.route, r.destinationairport ) as route,
+        recroute.depth+1 as depth
+    FROM `travel-sample`.inventory.route r JOIN recroute ON 
+        recroute.destinationairport = r.sourceairport
+    WHERE r.sourceairport!="TLV" and recroute.depth<2 
+)
+SELECT * FROM recroute LIMIT 10;
```

Comment:
LIMIT BY 10 defeats the purpose of the example as, all results would indicate traversal of depth 0 or only the root of the graph

After trying again I can confirm that it takes atleast 2000docs to see results from an iteration of depth = 2.

Which is why I would show simpler examples shown in the dev google doc.
But add the travel-sample example at the end just for reference.
Also those example indicate usage of the OPTIONS and CYCLE subclause.

you can see here how snowflake have done a similar job with a smaller example allowing user to first setup the data using insert and later a query to show usage.
https://docs.snowflake.com/en/user-guide/queries-cte#recursive-ctes-and-hierarchical-data